src/services/codefixes/inferFromUsage.ts

```diff
@@ -340,6 +376,27 @@ namespace ts.codefix {
         }
     }
 
+    function inferTypeForThisFromUsage(containingFunction: SignatureDeclaration, program: Program, cancellationToken: CancellationToken) {
+        let searchToken;
+
+        switch (containingFunction.kind) {
+            case SyntaxKind.FunctionExpression:
+                const parent = containingFunction.parent;
+                searchToken = isVariableDeclaration(parent) && isIdentifier(parent.name) ?
+                    parent.name :
+                    containingFunction.name;
+                break;
+            case SyntaxKind.FunctionDeclaration:
+            case SyntaxKind.MethodDeclaration:

```

Comment:
can you add a test for MethodDeclaration?

Reply:
I actually think this is not necessary, as `this` is inferred to be the container object:

```ts
interface Container {
    member: string;
    returnThisMember(): string;
}

const container: Container = {
    member: "sample",
    returnThisMember() {
        return this.member; // `this` is already `Container`
    },
};
```